### PR TITLE
[sentry-crashpad] Update to 0.6.1

### PR DESCRIPTION
**Pull request by MartinDelille**
---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/sentry-crashpad/all/conanfile.py
Comment (prince-chrismc) [inline recipes/sentry-crashpad/all/conanfile.py:98]: This should ideally be a patch file

**Review by prince-chrismc — APPROVED** [review]

File: recipes/sentry-crashpad/all/conanfile.py
Comment (MartinDelille) [reply recipes/sentry-crashpad/all/conanfile.py:98]: The problem is that it depends on the `with_tls` option. Is there a way to apply a patch conditionally ?

**Review by samuel-emrys — APPROVED** [review]

**Review by samuel-emrys — CHANGES_REQUESTED** [review]
Could we fix the 2.0 pipeline before merging? It looks like you might just need to make the following change:

```diff
- self.output.warn("Compiler is unknown. Assuming it supports C++14.")
+ self.output.warning("Compiler is unknown. Assuming it supports C++14.")
```

File: recipes/sentry-crashpad/all/conanfile.py
Comment (prince-chrismc) [reply recipes/sentry-crashpad/all/conanfile.py:98]: No! You are spot on

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
Seem's like CURL's required?


```
CMake Error at /opt/pyenv/versions/3.7.13/lib/python3.7/site-packages/cmake/data/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find CURL (missing: CURL_LIBRARY CURL_INCLUDE_DIR)
Call Stack (most recent call first):
  /opt/pyenv/versions/3.7.13/lib/python3.7/site-packages/cmake/data/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
  /opt/pyenv/versions/3.7.13/lib/python3.7/site-packages/cmake/data/share/cmake-3.18/Modules/FindCURL.cmake:169 (find_package_handle_standard_args)
  util/CMakeLists.txt:253 (find_package)
  ```

**Comment by prince-chrismc** [comment]
https://github.com/getsentry/crashpad/blob/2237d97ee2c38c930c07001e660be57324f69a37/util/CMakeLists.txt#L250

**Comment by samuel-emrys** [comment]
My read of the previous error message was that you might also need something like:

```diff
# util
        self.cpp_info.components["crashpad_util"].set_property("cmake_target_name", "crashpad::util")
        self.cpp_info.components["crashpad_util"].libs = ["crashpad_util"]
        self.cpp_info.components["crashpad_util"].requires = ["crashpad_compat", "crashpad_mini_chromium", "zlib::zlib"]
        if self.settings.os in ("Linux", "FreeBSD"):
            self.cpp_info.components["crashpad_util"].system_libs.extend(["pthread", "rt"])
+           self.cpp_info.components["crashpad_util"].requires.extend(["CURL::libcurl"])
```

**Comment by MartinDelille** [comment]
The last commit was just a try but could be reverted I guess. Any idea what is going wrong here ?

**Comment by samuel-emrys** [comment]
This is the patch that got this "working" for me:

```diff
diff --git a/recipes/sentry-crashpad/all/conanfile.py b/recipes/sentry-crashpad/all/conanfile.py
index e14673a28..4652a04db 100644
--- a/recipes/sentry-crashpad/all/conanfile.py
+++ b/recipes/sentry-crashpad/all/conanfile.py
@@ -143,7 +143,7 @@ class SentryCrashpadConan(ConanFile):
         self.cpp_info.components["crashpad_util"].requires = ["crashpad_compat", "crashpad_mini_chromium", "zlib::zlib"]
         if self.settings.os in ("Linux", "FreeBSD"):
             self.cpp_info.components["crashpad_util"].system_libs.extend(["pthread", "rt"])
-            self.cpp_info.components["crashpad_util"].requires.extend(["CURL::libcurl"])
+            self.cpp_info.components["crashpad_util"].requires.extend(["libcurl::libcurl"])
         elif self.settings.os == "Windows":
             self.cpp_info.components["crashpad_util"].system_libs.append("winhttp")
         elif self.settings.os == "Macos":
diff --git a/recipes/sentry-crashpad/all/test_package/conanfile.py b/recipes/sentry-crashpad/all/test_package/conanfile.py
index 3a9d92c6b..5e9dd17e4 100644
--- a/recipes/sentry-crashpad/all/test_package/conanfile.py
+++ b/recipes/sentry-crashpad/all/test_package/conanfile.py
@@ -27,5 +27,5 @@ class TestPackageConan(ConanFile):
             mkdir(self, test_env_dir)
             bin_path = os.path.join(self.cpp.build.bindirs[0], "test_package")
             handler_exe = "crashpad_handler.exe" if self.settings.os == "Windows" else "crashpad_handler"
-            handler_bin_path = os.path.join(self.deps_cpp_info["sentry-crashpad"].bin_paths[0], handler_exe)
-            self.run(f"{bin_path} {test_env_dir} {handler_bin_path}", run_environment=True)
+            handler_bin_path = os.path.join(self.dependencies["sentry-crashpad"].cpp_info.bindirs[0], handler_exe)
+            self.run(f"{bin_path} {test_env_dir} {handler_bin_path}", env="conanrun")
```

Note that specifically for the `test_package`, this isn't conan 1.x and 2.0 compatible. In 1.x, `self.dependencies` isn't available in the `test()` method. To make this compatible for both you'll need to do this in the `generate()` method and save it to json or something, then read it back in in `test()`.

I'm also not sure if `crashpad_util` is the correct component to model the dependency on `libcurl`. I'm not familiar with this library at all and a cursory inspection of the root CMakeLists.txt shows that `libcurl` is a dependency of the `sentry` target, which isn't modelled in the conan representation, so it's not clear to me that this is right - i'll leave that to people familiar with the library.

It's also not clear to me that the test package is working _correctly_ - it certainly executes and finishes with a non-error exit code but there's no output, and the command executed looks a little funny:

```
sentry-crashpad/0.6.1@user/testing (test package): RUN: ./test_package test_env /home/user/.conan2/p/sentr5675c4e3d1df6/p/bin/crashpad_handler
```

But again - not being familiar with the library I can't say if this is normal or not without looking into it more.

**Comment by MartinDelille** [comment]
Thanks @samuel-emrys ! I just corrected the component for now because the tests used to work without issue before my change. I'll look more into it if it still fails.

**Review by prince-chrismc — APPROVED** [review]
Let's get this is, we can fix v2 in the next one

**Review by SSE4 — APPROVED** [review]

**Pull request merged**
